core/src/main/kotlin/supplier/CacheAwareRestSupplier.kt

```diff
@@ -0,0 +1,171 @@
+package dev.kord.core.supplier
+
+import dev.kord.cache.api.DataCache
+import dev.kord.cache.api.put
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.entity.*
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.GuildChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+/**
+ * [EntitySupplier] that uses a [RestEntitySupplier] to resolve entities.
+ *
+ * Resolved entities will always be stored in [cache] if it wasn't null or empty for flows.
+ */
+class CacheAwareRestEntitySupplier(
+    val restSupplier: RestEntitySupplier,
```

Comment:
is it possible to lower the requirement to any supplier? It would make this a more flexible solution.

Reply:
Hmmm, yeah sounds like someone might want to use their own entity supplier and store in a data cache.
Into it as well!